tests/executables_src/testUserInputValidator.cc

```diff
@@ -611,15 +613,39 @@ BOOST_AUTO_TEST_CASE(testDeepBackwardsPropagation) {
   BOOST_TEST(upstrIn == 5);
 
   // test two consecutive values, only the first being discarded at the lowest level and the second is accepted
+repeat:
   upstrIn.setAndWrite(12);
   upstrIn.setAndWrite(3);
 
   test.stepApplication();
-  BOOST_TEST(downstrIn.readLatest()); // just observe final state, because intermediate states might be subject
-  BOOST_TEST(downstrIn == 5);         // to race conditions
+
+  // There are two acceptable outcomes of this test:
+  //
+  // 1) Likely: The first value was rejected and the second was accepted
+  //
+  // 2) Unlikely: Both values are rejected. This can happen because the UserInputValidator of midstream needs to use a
+  //    fresh VersionNumber to propagate the rejection of the first value from downstream (otherwise the scenario in
+  //    testBackwardsPropagationTwoDownstream would break). Since that fresh VersionNumber is bigger then the one of the
+  //    second, valid value, it can overwrite the second value and hence that gets effectively rejected.
+  //
+  // Currently, we accept both scenarios but require that the likely scenario is observed (by retrying a couple of times
+  // if we see the second scenario). This problem can be solved by extending the VersionNumber with a sub-version, so we
+  // can both distinguish the corrected from the rejected values as well as find out to which original VersionNumber
+  // the corrected value belongs. Due to the fact that this problem only occurs when writing inputs faster than the
+  // values get rejected and the UserInputValidator being designed for inputs by users, this problem does not seem to
+  // play a big role in real scenarios.
+
+  BOOST_TEST(downstrIn.readLatest()); // just observe final state, to avoid intermediate races
   BOOST_TEST(midstreamIn.readLatest());
-  BOOST_TEST(midstreamIn == 4);
+  if(downstrIn == 7) {
+    BOOST_TEST(upstrIn.readLatest());
+    BOOST_TEST(midstreamIn == 6);
+    BOOST_TEST(upstrIn == 5);
+    goto repeat;
```

Comment:
Do you want to put a timeout/ maxRetries? This has the potential to hang forever.

However, the timeouts in DeviceAccess are currently failing after very long times...

Reply:
done